Projects/TerminateContracts/Sources/Views/TerminationSummaryScreen.swift

```diff
@@ -3,9 +3,29 @@ import hCore
 import hCoreUI
 
 struct TerminationSummaryScreen: View {
-    @EnvironmentObject var terminationNavigationVm: TerminationFlowNavigationViewModel
+    @ObservedObject var terminationNavigationVm: TerminationFlowNavigationViewModel
+    let extraCoverageItems: [ExtraCoverageItem]?
+    let withAddonView: Bool
 
-    init() {}
+    init(
+        terminationNavigationVm: TerminationFlowNavigationViewModel
+    ) {
+        self.terminationNavigationVm = terminationNavigationVm
+
+        self.extraCoverageItems = {
```

Comment:
Can we have this logic inside TerminationFlowNavigationViewModel ?
like:
`    @Published var extraCoverage: [ExtraCoverageItem] = []
`
and just update
```
    @Published var terminationDateStepModel: TerminationFlowDateNextStepModel? {
        didSet{
            extraCoverage = terminationDateStepModel?.extraCoverageItem ?? []
        }
    }
    
    @Published var terminationDeleteStepModel: TerminationFlowDeletionNextModel?{
        didSet{
            extraCoverage = terminationDeleteStepModel?.extraCoverageItem ?? []
        }
    }
```

Which should enable you to remove
```
    let extraCoverageItems: [ExtraCoverageItem]?
    let withAddonView: Bool
```
here